dev/com.ibm.ws.jsp.2.3_fat/fat/src/com/ibm/ws/jsp23/fat/tests/JSPJava7Test.java

```diff
@@ -0,0 +1,172 @@
+    @Mode(TestMode.FULL)
+    public void testJdkSourceLevel17() throws Exception {
+
+        ServerConfiguration configuration = server.getServerConfiguration();
+        configuration.getJspEngine().setJdkSourceLevel("17");
+        configuration.getJspEngine().setUseJDKCompiler(false);
+        LOG.info("New server configuration used: " + configuration);
+
+        server.updateServerConfiguration(configuration);
+        server.restartServer();
+
+        WebConversation wc = new WebConversation();
+        wc.setExceptionsThrownOnErrorStatus(false);
+
+        String url = JSPUtils.createHttpUrlString(server, APP_NAME, "index.jsp");
+        LOG.info("url: " + url);
+
+        WebRequest request = new GetMethodWebRequest(url);
+        WebResponse response = wc.getResponse(request);
+        LOG.info("Servlet response : " + response.getText());
+
+        assertEquals("Expected " + 500 + " status code was not returned!",
+                     500, response.getResponseCode());
+
+        assertTrue("Compilation error message not found!", response.getText().contains("Lambda expressions are allowed only at source level 1.8 or above")); 
+    }
+
+    /*
+     * Ensure JdkSourceLevel works by testing option 17 (Java 7).
+     * Lambda were added in Java 8, so the JSP is expected to fail.
+     */
+    @Test
+    @AllowedFFDC("java.security.PrivilegedActionException") // Occurs in EE10 and lower
+    @Mode(TestMode.FULL)
+    public void testJdkSourceLevel17withPreCompile() throws Exception {
+
+        ServerConfiguration configuration = server.getServerConfiguration();
+        configuration.getJspEngine().setJdkSourceLevel("17");
+        configuration.getJspEngine().setUseJDKCompiler(false);
+        configuration.getJspEngine().setPrepareJSPs("0");
+        LOG.info("New server configuration used: " + configuration);
+
+        server.updateServerConfiguration(configuration);
+        server.stopServer("SRVE8115W","SRVE8094W");
+        server.startServer();
+
+        WebConversation wc = new WebConversation();
+        wc.setExceptionsThrownOnErrorStatus(false);
+
+        String url = JSPUtils.createHttpUrlString(server, APP_NAME, "index.jsp");
+        LOG.info("url: " + url);
+
+        WebRequest request = new GetMethodWebRequest(url);
+        WebResponse response = wc.getResponse(request);
+        LOG.info("Servlet response : " + response.getText());
+
+        assertEquals("Expected " + 500 + " status code was not returned!",
+                     500, response.getResponseCode());
+
+        assertTrue("Compilation error message not found!", response.getText().contains("Lambda expressions are allowed only at source level 1.8 or above")); 
+    }
+
+    /*
+     * Ensure JdkSourceLevel works by testing option 17 (Java 7).
+     * Lambda were added in Java 8, so the JSP is expected to fail.
+     */
+    @Test
+    @AllowedFFDC("java.security.PrivilegedActionException") // Occurs in EE10 and lower
```

Comment:
What is the exception something in JSP we need to take a look at in the future or is this one expected?

Reply:
```
Exception = java.security.PrivilegedActionException
Source = com.ibm.ws.jsp.webcontainerext.AbstractJSPExtensionProcessor.findWrapper
probeid = 445
Stack Dump = java.security.PrivilegedActionException: com.ibm.ws.jsp.JspCoreException: JSPG0049E: /index.jsp failed to compile : 

JSPG0091E: An error occurred at line: 27 in the file: /index.jsp
JSPG0093E: Generated servlet error from file: /index.jsp 

/Users/volosied/libertyGit/open-liberty/dev/build.image/wlp/usr/servers/jspJava7Server/workarea/org.eclipse.osgi/76/data/temp/default_node/SMF_WebContainer/TestJSPWithJava8/TestJSPWithJava8/_index.java:139: error: lambda expressions are not supported in -source 7
          messages.stream().forEach(e -> write.print(e));
                                      ^
  (use -source 8 or higher to enable lambda expressions)

	at java.base/java.security.AccessController.doPrivileged(AccessController.java:573)
	at com.ibm.ws.jsp.webcontainerext.AbstractJSPExtensionProcessor.findWrapper(AbstractJSPExtensionProcessor.java:482)
	at com.ibm.ws.jsp.webcontainerext.ws.PrepareJspHelper$PrepareJspHelperThread.handleCompile(PrepareJspHelper.java:493)
	at com.ibm.ws.jsp.webcontainerext.ws.PrepareJspHelper$PrepareJspHelperThread.run(PrepareJspHelper.java:460)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: com.ibm.ws.jsp.JspCoreException: JSPG0049E: /index.jsp failed to compile : 
```

Not sure why it's a security exception, but the compilation is the cause.